source/adios2/CMakeLists.txt

```diff
@@ -247,7 +247,20 @@ if (ADIOS2_HAVE_SST)
   target_link_libraries(adios2_core PRIVATE adios2::thirdparty::EVPath)
   add_subdirectory(toolkit/remote)
 endif()
-   
+
+if (ADIOS2_HAVE_KVCACHE)
+  target_sources(adios2_core PRIVATE  toolkit/cache/KVCacheCommon.h
+    toolkit/cache/KVCacheCommon.tcc)
+  target_link_libraries(adios2_core PRIVATE hiredis::hiredis)
+
+  # Ensure the rpath is set correctly
+  # message(STATUS "PACKAGE_PREFIX_DIR: ${PACKAGE_PREFIX_DIR}")
+  set_target_properties(adios2_core PROPERTIES
+    BUILD_RPATH "$ORIGIN:${PACKAGE_PREFIX_DIR}/lib"
+    INSTALL_RPATH "$ORIGIN:${PACKAGE_PREFIX_DIR}/lib"
```

Comment:
Why is this needed?

Reply:
Otherwise, in '~/project/ADIOS2-Source$ ldd build-cache-test/bin/adios2_remote_server', it will show:
libhiredis.so.1.1.0 =>  not found

and user needs to manually set LD_LIBRARY_PATH by 'export LD_LIBRARY_PATH=${SW_DIR}/adios2/lib:${SW_DIR}/hiredis/lib:$LD_LIBRARY_PATH'

Reply:
${PACKAGE_PREFIX_DIR} refers to hiredis installation path and set in hiredis cmake.

Reply:
I understand the need of rpath, what I do not is why setting it manually. In cmake this is handled by cmake global cmake flags, you do not need to set it yourself unless there is a very particular issue.

The thing is that you do not set the rpath to your particular external dependency, you use other means for this (LD_library_path is the slowest one).

The problem with this is that doing that is not portable since in a binary distribution this assumption of where to find the particular library will fail since the user might have its library in a different dir than the directory that was uses when building adios2 (eg: a package manager case).

I am afraid that you will either need to setup correctly your LD_LIBRARY_PATH or install your dependencies in a place that the linker can find it.

Reply:
Ok, I figure it out by passing -DCMAKE_PREFIX_PATH="${SW_DIR}/hiredis" and updated.

Reply:
I think I met a weird problem when I use -DCMAKE_PREFIX_PATH="${SW_DIR}/hiredis". The link under bin works fine, but under lib is not found. Do you know how to fix it?

4cv@PC0141240:~/project/ADIOS2-Source/examples/simulations/gray-scott/build$ ldd /home/4cv/project/ADIOS2-Source/build/sw/adios2/lib/libadios2_core.so
	linux-vdso.so.1 (0x00007ffdb4ffc000)
	libhiredis.so.1.1.0 => not found
	libsqlite3.so.0 => /lib/x86_64-linux-gnu/libsqlite3.so.0 (0x000076b73ddbc000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x000076b73d5e4000)
	libadios2_perfstubs.so.2.10 => /home/4cv/project/ADIOS2-Source/build/sw/adios2/lib/../lib/libadios2_perfstubs.so.2.10 (0x000076b73ddb6000)
	libadios2_evpath.so.2.10 => /home/4cv/project/ADIOS2-Source/build/sw/adios2/lib/../lib/libadios2_evpath.so.2.10 (0x000076b73d569000)
	libadios2_ffs.so.2.10 => /home/4cv/project/ADIOS2-Source/build/sw/adios2/lib/../lib/libadios2_ffs.so.2.10 (0x000076b73d4fe000)
	libadios2_atl.so.2.10 => /home/4cv/project/ADIOS2-Source/build/sw/adios2/lib/../lib/libadios2_atl.so.2.10 (0x000076b73d4ef000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x000076b73d200000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x000076b73d119000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x000076b73d4cf000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x000076b73ce00000)
	/lib64/ld-linux-x86-64.so.2 (0x000076b73df1d000)
	libadios2_dill.so.2.10 => /home/4cv/project/ADIOS2-Source/build/sw/adios2/lib/../lib/../lib/libadios2_dill.so.2.10 (0x000076b73d47a000)


4cv@PC0141240:~/project/ADIOS2-Source/examples/simulations/gray-scott/build$ ldd /home/4cv/project/ADIOS2-Source/build/bin/adios2_remote_server 
	linux-vdso.so.1 (0x00007fff52fc0000)
	libadios2_evpath.so.2.10 => /home/4cv/project/ADIOS2-Source/build/lib/libadios2_evpath.so.2.10 (0x000077ab8af0f000)
	libadios2_core.so.2.10 => /home/4cv/project/ADIOS2-Source/build/lib/libadios2_core.so.2.10 (0x000077ab8a600000)
	libadios2_atl.so.2.10 => /home/4cv/project/ADIOS2-Source/build/lib/libadios2_atl.so.2.10 (0x000077ab8af00000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x000077ab8a200000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x000077ab8ae07000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x000077ab8ade5000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x000077ab89e00000)
	libadios2_ffs.so.2.10 => /home/4cv/project/ADIOS2-Source/build/lib/libadios2_ffs.so.2.10 (0x000077ab8a595000)
	libhiredis.so.1.1.0 => /home/4cv/project/ADIOS2-Source/build/sw/hiredis/lib/libhiredis.so.1.1.0 (0x000077ab8adcd000)
	libsqlite3.so.0 => /lib/x86_64-linux-gnu/libsqlite3.so.0 (0x000077ab8a448000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x000077ab8adb1000)
	libadios2_perfstubs.so.2.10 => /home/4cv/project/ADIOS2-Source/build/lib/libadios2_perfstubs.so.2.10 (0x000077ab8a442000)
	/lib64/ld-linux-x86-64.so.2 (0x000077ab8afd6000)
	libadios2_dill.so.2.10 => /home/4cv/project/ADIOS2-Source/build/lib/libadios2_dill.so.2.10 (0x000077ab8a1ab000)